scripts/hail_batch/hgdp1kg_tobwgs_nfe/hgdp_1kg_tob_wgs_nfe.py

```diff
@@ -0,0 +1,80 @@
+"""Perform pca on nfe samples from the HGDP,1KG, and tob-wgs dataset"""
+
+import click
+import pandas as pd
+import hail as hl
+from hail.experimental import lgt_to_gt
+
+HGDP1KG_TOBWGS = (
+    'gs://cpg-tob-wgs-analysis/1kg_hgdp_tobwgs_pca/'
+    'v1/hgdp1kg_tobwgs_joined_all_samples.mt'
+)
+
+GNOMAD_LIFTOVER_LOADINGS = 'gs://cpg-reference/gnomad/gnomad_loadings_90k_liftover.ht'
+
+GNOMAD_HGDP_1KG_MT = (
+    'gs://gcp-public-data--gnomad/release/3.1/mt/genomes/'
+    'gnomad.genomes.v3.1.hgdp_1kg_subset_dense.mt'
+)
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/joint_vcf/v1/raw/genomes.mt'
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    mt = hl.read_matrix_table(HGDP1KG_TOBWGS)
+    # Get NFE samples only
+    mt = mt.filter_cols(
+        (mt.hgdp_1kg_metadata.population_inference.pop == 'nfe')
+        | (mt.s.contains('TOB'))
+    )
+
+    # Perform PCA
+    eigenvalues_path = f'{output}/eigenvalues.csv'
+    scores_path = f'{output}/scores.ht'
+    loadings_path = f'{output}/loadings.ht'
+    eigenvalues, scores, loadings = hl.hwe_normalized_pca(
+        mt.GT, compute_loadings=True, k=20
+    )
+    eigenvalues_df = pd.DataFrame(eigenvalues)
+    eigenvalues_df.to_csv(eigenvalues_path, index=False)
+    scores.write(scores_path, overwrite=True)
+    loadings.write(loadings_path, overwrite=True)
+
+    # get TOB-WGS allele frequencies
+    tob_wgs = hl.read_matrix_table(
+        'gs://cpg-tob-wgs-temporary/joint_vcf/v1/raw/genomes.mt'
+    ).key_rows_by('locus', 'alleles')
+    tob_wgs = tob_wgs.annotate_entries(GT=lgt_to_gt(tob_wgs.LGT, tob_wgs.LA))
+    tob_wgs = tob_wgs.annotate_rows(
+        gt_stats=hl.agg.call_stats(tob_wgs.GT, tob_wgs.alleles)
+    )
+
+    # Get gnomAD allele frequency of variants that aren't in TOB-WGS
+    loadings_gnomad = hl.read_table(GNOMAD_LIFTOVER_LOADINGS).key_by('locus', 'alleles')
+    hgdp_1kg = hl.read_matrix_table(GNOMAD_HGDP_1KG_MT)
+    loadings_gnomad = loadings_gnomad.annotate(
+        gnomad_AF=hgdp_1kg.rows()[
+            loadings_gnomad.locus, loadings_gnomad.alleles
+        ].gnomad_freq.AF,
+        gnomad_popmax_AF=hgdp_1kg.rows()[
+            loadings_gnomad.locus, loadings_gnomad.alleles
+        ].gnomad_popmax.AF,
+        TOB_variant=hl.is_defined(
+            mt.rows()[loadings_gnomad.locus, loadings_gnomad.alleles].gnomad_popmax.AF
+        ),
+        TOB_WGS_AF=tob_wgs.rows()[
+            loadings_gnomad.locus, loadings_gnomad.alleles
+        ].gt_stats.AF,
+    )
```

Comment:
- Do you need `TOB_variant`? Won't `TOB_WGS_AF` evaluate to `missing` for any variant that's not defined?
- Do you also want to copy the `gnomad_freq_meta` description, so we know what `gnomad_freq.AF` refers to in this table?
- `loadings` is kind of a misleading name here, as we're really only interested in the SNVs and their corresponding frequencies. Can we rename the variable / output path and remove the actual loadings values?

```suggestion
    hgdp_1kg_row = hgdp_1kg.rows()[loadings_gnomad.locus, loadings_gnomad.alleles]
    tob_wgs_row = tob_wgs.rows()[loadings_gnomad.locus, loadings_gnomad.alleles]
    loadings_gnomad = loadings_gnomad.annotate(
        gnomad_AF=hgdp_1kg_row.gnomad_freq.AF,
        gnomad_popmax_AF=hgdp_1kg_row.gnomad_popmax.AF,
        TOB_WGS_AF=tob_wgs_row.gt_stats.AF,
    )
```

Reply:
Good ideas. I copied the `gnomad_freq_meta` global field, removed `TOB_variant`, and renamed the loadings mt.